pkg/scheduler/internal/queue/scheduling_queue_test.go

```diff
@@ -359,6 +360,73 @@ func TestPriorityQueue_Delete(t *testing.T) {
 	}
 }
 
+func BenchmarkMoveAllToActiveOrBackoffQueue(b *testing.B) {
+	tests := []struct {
+		name string
+		// Init pods in activeQ and unschedulableQ respectively.
+		podsInActiveQ, podsInUnschedulableQ int
+	}{
+		{
+			name:                 "1k-pods",
+			podsInActiveQ:        200,
+			podsInUnschedulableQ: 800,
+		},
+		{
+			name:                 "5k-pods",
+			podsInActiveQ:        1000,
+			podsInUnschedulableQ: 4000,
+		},
+	}
+
+	podTemplates := []v1.Pod{highPriorityPod, highPriNominatedPod, medPriorityPod, unschedulablePod}
+
+	for _, tt := range tests {
+		b.Run(tt.name, func(b *testing.B) {
+			for i := 0; i < b.N; i++ {
+				b.StopTimer()
+				c := clock.NewFakeClock(time.Now())
+				q := NewPriorityQueue(newDefaultQueueSort(), WithClock(c))
+
+				// Init pods in activeQ.
+				for i := 0; i < tt.podsInActiveQ; i++ {
+					p := podTemplates[i%len(podTemplates)].DeepCopy()
+					p.Name, p.UID = fmt.Sprintf("%v-%v", p.Name, i), types.UID(fmt.Sprintf("%v-%v", p.UID, i))
+					q.Add(p)
+				}
+				// Init pods in unschedulableQ.
+				for i := 0; i < tt.podsInUnschedulableQ; i++ {
+					p := podTemplates[i%len(podTemplates)].DeepCopy()
+					p.Name, p.UID = fmt.Sprintf("%v-%v", p.Name, i+tt.podsInActiveQ), types.UID(fmt.Sprintf("%v-%v", p.UID, i+tt.podsInActiveQ))
+					podInfo := q.newQueuedPodInfo(p)
+					podInfo.Attempts = 1
+					q.AddUnschedulableIfNotPresent(podInfo, q.SchedulingCycle())
+				}
+
+				// We cannot use wait.Until() as it wraps a RealClock.
+				go wait.BackoffUntil(q.flushBackoffQCompleted, wait.NewJitteredBackoffManager(1.0*time.Second, 0.0, c), true, q.stop)
+
+				b.StartTimer()
+				// Iterate 10 times, each time we step 1 second forward.
+				for j := 0; j < 10; j++ {
+					c.Step(time.Second)
+					q.MoveAllToActiveOrBackoffQueue("test")
```

Comment:
what if we just run this once, and there is no re-queuing. That should be enough for a unit benchmark. We don't even need pods in the active queue.

The point of this benchmark is to prove that the overhead of matching events is lower than moving all pods to the active queue.

Reply:
> The point of this benchmark is to prove that the overhead of matching events is lower than moving all pods to the active queue.

And even if it is not, this doesn't mean we shouldn't keep it. The optimization is not coming from lowering the cost of this specific function, it is from saving unnecessary scheduling cycles, which we can't measure using this benchmark.

Reply:
The point of the benchmark is to make sure that the cost is not too significant in the worst case.

Reply:
I got @ahg-g 's point. I'd like to use the unit benchmark test to focus on podMatchesEvent, while using integration benchmark to demonstrate the perf delta in an e2e flow.

So I will close this PR and amend a new UT benchmark test as a second commit in #98241. @ahg-g @alculquicondor please review in #98241.